### PR TITLE
Fixing uping

### PR DESCRIPTION
**Pull request by nornes**
lots of minor improvements

### CODE REVIEW

File: GATEWAY/src/mocks.ts [hunk 1]
```diff
@@ -166,3 +166,10 @@ export class DiagnosticMock {
     return true;
   }
 }
+
+export class ViewControllerMock {
```
Comment (TheFakeOG) [inline GATEWAY/src/mocks.ts]: These are not necessary

File: GATEWAY/src/providers/tilesApi.service.spec.ts [hunk 2]
```diff
@@ -3,9 +3,9 @@ import { Http, Response, ResponseOptions, BaseRequestOptions, RequestMethod } fr
 import { Storage } from '@ionic/storage';
 import { MockBackend, MockConnection } from '@angular/http/testing';
 import { Application, Device, LoginData, VirtualTile } from './utils.service';
-import { App, Config, AlertController, Platform } from 'ionic-angular';
+import { App, Config, AlertController, Platform, ViewController } from 'ionic-angular';
```
Comment (TheFakeOG) [inline GATEWAY/src/providers/tilesApi.service.spec.ts]: Not necessary

File: GATEWAY/src/providers/tilesApi.service.spec.ts [hunk 3]
```diff
@@ -3,9 +3,9 @@ import { Http, Response, ResponseOptions, BaseRequestOptions, RequestMethod } fr
...
 import { TilesApi } from './tilesApi.service';
-import { StorageMock } from '../mocks';
+import { StorageMock, ViewControllerMock } from '../mocks';
```
Comment (TheFakeOG) [inline GATEWAY/src/providers/tilesApi.service.spec.ts]: Not necessary

File: GATEWAY/src/providers/tilesApi.service.spec.ts [hunk 4]
```diff
@@ -39,6 +39,10 @@ describe('tilesAPI', () => {
           provide: Storage,
           useClass: StorageMock
         },
+        {
+          provide: ViewController,
+          useClass: ViewControllerMock
+        },
```
Comment (TheFakeOG) [inline GATEWAY/src/providers/tilesApi.service.spec.ts]: Not necessary

File: GATEWAY/src/providers/tilesApi.service.ts [hunk 5]
```diff
@@ -183,21 +183,37 @@ export class TilesApi {
    * @param {string} virtualTileId - The virtual tile
    * @param {string} applicationId - The application the virtual tile is registered to
    */
-  public pairDeviceToVirualTile = (deviceId: string, virtualTileId: string): Promise<Response> => {
+  public pairDeviceToVirualTile = (deviceId: string, virtualTileId: string): Promise<void> => {
```
Comment (TheFakeOG) [inline GATEWAY/src/providers/tilesApi.service.ts:186]: Why change this?

File: GATEWAY/src/providers/ble.service.spec.ts [hunk 6]
```diff
@@ -5,11 +5,11 @@ import { Storage } from '@ionic/storage';
 import { BackgroundFetch } from '@ionic-native/background-fetch';
 import { BLE } from '@ionic-native/ble';
 import { Diagnostic } from '@ionic-native/diagnostic';
-import { App, Config, AlertController, Platform, Events } from 'ionic-angular';
+import { App, Config, AlertController, Platform, Events, ViewController } from 'ionic-angular';
```
Comment (TheFakeOG) [inline GATEWAY/src/providers/ble.service.spec.ts]: Not necessary

File: GATEWAY/src/providers/ble.service.spec.ts [hunk 7]
```diff
@@ -5,11 +5,11 @@ import { Storage } from '@ionic/storage';
...
 import { Observable, Subscription } from 'rxjs';
 import { BleService } from './ble.service';
 import { DevicesService }from './devices.service';
-import { StorageMock, BackgroundFetchMock, BLEMock, DiagnosticMock } from '../mocks';
+import { StorageMock, BackgroundFetchMock, BLEMock, DiagnosticMock, ViewControllerMock } from '../mocks';
```
Comment (TheFakeOG) [inline GATEWAY/src/providers/ble.service.spec.ts]: Not necessary

File: GATEWAY/src/providers/ble.service.spec.ts [hunk 8]
```diff
@@ -35,6 +35,10 @@ describe('bleService', () => {
         BleService,
         AlertController,
         {
+        provide: ViewController,
+        useClass: ViewControllerMock
+        },
```
Comment (TheFakeOG) [inline GATEWAY/src/providers/ble.service.spec.ts]: Not necessary

**Review by TheFakeOG — CHANGES_REQUESTED** [review]
All these are not necessary
I have already implemented a solution that works 👍

File: GATEWAY/src/providers/tilesApi.service.ts (hunk 5 above)
Comment (nornes) [reply GATEWAY/src/providers/tilesApi.service.ts:186]: because it returns a void, not a response

File: GATEWAY/src/mocks.ts (hunk 1 above)
Comment (nornes) [reply GATEWAY/src/mocks.ts]: you added them, so if they weren't supposed to be there I feel you should be the one removing them 😉

File: GATEWAY/src/providers/tilesApi.service.ts (hunk 5 above)
Comment (TheFakeOG) [reply GATEWAY/src/providers/tilesApi.service.ts:186]: Actually it does return a response, and it is the response from the http.post
That is why that test breaks
I have currently hacked the test to ignore it and pass the test no matter what it gets back, but i would prefer if we could leave it as it was

File: GATEWAY/src/mocks.ts (hunk 1 above)
Comment (TheFakeOG) [reply GATEWAY/src/mocks.ts]: I did remove them
Did you pull my last changes before the PR?

File: GATEWAY/src/providers/tilesApi.service.ts (hunk 5 above)
Comment (TheFakeOG) [reply GATEWAY/src/providers/tilesApi.service.ts:186]: It does not really matter which one it is, as the tests are set up to pass at the moment
So it is up to you 👍

Comment (nornes) [reply GATEWAY/src/providers/tilesApi.service.ts:186]: it used to return a response, but now it doesn't anymore. I can make it return the result, but we do not need it for anything else

Comment (nornes) [reply GATEWAY/src/providers/tilesApi.service.ts:186]: You can test that it was successfull by spying on the method beeing called in the then statement

Comment (TheFakeOG) [reply GATEWAY/src/providers/tilesApi.service.ts:186]: Then we'll just dismiss this review

**Review by TheFakeOG — APPROVED** [review]

File: GATEWAY/src/providers/tilesApi.service.ts (hunk 5 above)
Comment (TheFakeOG) [reply GATEWAY/src/providers/tilesApi.service.ts:186]: Or approve 😛

**Review by aaursand — APPROVED** [review]
looks good to me

**Pull request merged**